### DISCUSSION TITLE
"Info" Topic for MQTT

### DISCUSSION THREAD

**Discussion by drhirn** [Hypfer/Valetudo#1644] [Q&A / Support]
For integrating Valetudo via MQTT into other Smart Home Systems than Home Assistant I could use a "info"-topic with some basic information about Valetudo and the robot. Like
- Valetudo release
- robot model
- robot ip
 And it would be great if you could request this information via a "set" command.

Do others think this would be usefull?

**Comment by Hypfer** [comment]
> robot ip

https://valetudo.cloud/pages/integrations/mqtt.html#ipaddressesips

Publishing the Valetudo version and robot model on connect could be useful, yeah

**Reply by drhirn** [reply]
> > robot ip
> 
> https://valetudo.cloud/pages/integrations/mqtt.html#ipaddressesips

I know. But the ip get's never published, when I need it ;). That's why the ability to request this info would be useful.

To get more detailed: When adding a robot to FHEM, there's the need for this info to determine which model and which Valetudo release is used.

**Reply by Hypfer** [reply]
Sounds like your MQTT setup might be configured incorrectly.

As the retain flag is set for all those messages, you should not have to wait for any of those to be published.
It should be available as soon as you subscribe.

**Reply by drhirn** [reply]
True. But - as far as I actually know - there's no way to get retained topics in FHEM until they are re-published again.
